src/argilla/client/workspaces.py

```diff
@@ -13,68 +12,305 @@
+        except BaseClientError as e:
+            raise RuntimeError(
+                f"Error while adding user with id=`{user_id}` to workspace with" f" id=`{self.id}`."
+            ) from e
+
+    def delete_user(self, user_id: str) -> None:
+        """Deletes an existing user from the workspace in Argilla. Note that the user
+        will not be deleted from Argilla, but just from the workspace.
+
+        Args:
+            user_id: the ID of the user to be deleted from the workspace. The user must exist in Argilla.
 
-        if self.id is None:
-            warnings.warn(
-                f"The `rg.Workspace` doesn't exist yet, you need to create it in advance. Otherwise, use `rg.Workspace.from_name` to make sure that the workspace exists."
+        Raises:
+            ValueError: if the user with the provided ID doesn't exist in the workspace.
+            RuntimeError: if there was an error while deleting the user from the workspace.
+
+        Examples:
+            >>> from argilla import rg
+            >>> workspace = rg.Workspace.from_name("my-workspace")
+            >>> workspace.delete_user("my-user-id")
+        """
+        try:
+            workspaces_api.delete_workspace_user(
+                client=self.__client,
+                id=self.id,
+                user_id=user_id,
             )
+        except NotFoundApiError as e:
+            raise ValueError(
+                f"Either the user with id=`{user_id}` doesn't exist in Argilla, or it"
+                f" doesn't belong to workspace with id=`{self.id}`."
+            ) from e
+        except BaseClientError as e:
+            raise RuntimeError(
+                f"Error while deleting user with id=`{user_id}` from workspace with" f" id=`{self.id}`."
+            ) from e
+
+    @staticmethod
+    def __active_client() -> "httpx.Client":
+        """Returns the active Argilla `httpx.Client` instance."""
+        try:
+            return active_client().http_client.httpx
+        except Exception as e:
+            raise RuntimeError(f"The `rg.active_client()` is not available or not respoding.") from e
+
+    @classmethod
+    def create(cls, name: str) -> "Workspace":
+        """Creates a new workspace in Argilla.
+
+        Args:
+            name: the name of the workspace to be created.
+
+        Raises:
+            ValueError: if the workspace with the provided name already exists.
+
+        Returns:
+            A `Workspace` instance.
+
+        Examples:
+            >>> from argilla import rg
+            >>> workspace = rg.Workspace.create("my-workspace")
+        """
+        instance = cls.__new__(cls)
+        instance.__client = cls.__active_client()
+        try:
+            ws = workspaces_api.create_workspace(client=instance.__client, name=name).parsed
+            instance.__dict__.update(ws.dict())
+            return instance
```

Comment:
Maybe we can create a new instance after we're sure everything went well. 
```suggestion
        try:
            ws = workspaces_api.create_workspace(client=instance.__client, name=name).parsed
            # This code block is a potential WorspaceFactory method
            instance = cls.__new__(cls)
            instance.__client = cls.__active_client()
            instance.__dict__.update(ws.dict())
            return instance
```